### PR TITLE
add dockerfile

### PR DESCRIPTION
**Pull request by nlef**
add dockerfile and simple docker-compose

### CODE REVIEW

**Comment by nlef** [comment]
@moookino
pls take a look at this pr

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+RUN apt update \
+ && apt install -y \
+      python3-cryptography \
+      python3-gevent \
+      python3-opencv \
+      x264 \
+      libx264-dev \
+      libwebp-dev \
+      && apt clean \
+      && rm -rf /var/lib/apt/lists/*
```
Comment (mkuf) [inline Dockerfile:24]: `python:3.9-slim` contains everything you need to build the venv for your app. You may want to ditch the multi-stage build to speed up buildtime and simplify the image. 

```suggestion
FROM python:3.9-slim

RUN apt update \
 && apt install -y \
      python3-cryptography \
      python3-gevent \
      python3-opencv \
      x264 \
      libx264-dev \
      libwebp-dev \
 && apt clean \
 && rm -rf /var/lib/apt/lists/*
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+WORKDIR /opt
+
+RUN groupadd moonraker-telegram-bot --gid 1000 \
+ && useradd moonraker-telegram-bot --uid 1000 --gid moonraker-telegram-bot
+
+COPY --chown=moonraker-telegram-bot:moonraker-telegram-bot --from=build /opt/moonraker-telegram-bot ./moonraker-telegram-bot
+COPY --chown=moonraker-telegram-bot:moonraker-telegram-bot --from=build /opt/venv ./venv
+
+RUN mkdir -p printer_data/logs printer_data/config timelapse timelapse_finished \
+ && chown -R moonraker-telegram-bot:moonraker-telegram-bot /opt/*
```
Comment (mkuf) [inline Dockerfile:35]: When not using a multistage build, the venv does not have to be copied anymore and the copy command for the codebase needs to reference the build context.

Other then that, using chown for files created in a different layer, creates a new layer that takes up as much space as the one before, doubling the space used in the image. 

Merging the user/directory creation with the permission change creates only a single layer. The venv does not need to be owned by the user.

```suggestion
RUN groupadd moonraker-telegram-bot --gid 1000 \
 && useradd moonraker-telegram-bot --uid 1000 --gid moonraker-telegram-bot \
 && mkdir -p printer_data/logs printer_data/config timelapse timelapse_finished \
 && chown -R moonraker-telegram-bot:moonraker-telegram-bot /opt/*

COPY --chown=moonraker-telegram-bot:moonraker-telegram-bot . ./moonraker-telegram-bot
RUN python3 -m venv venv \
 && venv/bin/pip install -r moonraker-telegram-bot/scripts/requirements.txt
```

**Comment by nlef** [comment]
@mkuf 
Many thanks for the review and advice, I accepted the suggestions and updated the dockerfile.
I left the base image with the os distro specified - it’s easier to track the versions of binary packages and guarantee the workability)

**Comment by nlef** [comment]
@mkuf images with tag `development` are ready to use from https://hub.docker.com/r/lefskiy/moonraker-telegram-bot/tags
Upcomming bot release will add tag `latest` and version tags.

**Pull request merged**